src/main/java/ApplicationDriver.java

```diff
@@ -102,10 +93,34 @@ private static boolean handleQueryInput(String input) {
                 }
                 break;
             case "5":
-                System.out.println("Auto-scheduling is not yet implemented"); // TODO implement this functionality
+                printTasks();
+                Task task = chooseTask();
+                success = controller.suggestTimeToUser(task);
+                if (success) {
+                    System.out.println("Event created from task");
+                } else {
+                    System.out.println("Failed to create event from task");
+                }
                 break;
             case "6":
-                System.out.println("Manual-scheduling is not yet implemented"); // TODO implement this functionality
+                printTasks();
+                Task taskManual = chooseTask();
+                LocalDateTime userSuggestedTime;
+                boolean timeAvailable;
+
+                do {
+                    userSuggestedTime = inputTime();
+                    timeAvailable = controller.checkUserSuggestedTime(taskManual, userSuggestedTime);
+                } while (!timeAvailable);
```

Comment:
This block seems not parallel to the auto-scheduling, which calls a method instead using while loop here

Reply:
right, we could put the loop part in a controller class